src/app/CommandHandlerImpl.h

```diff
@@ -294,7 +294,13 @@ class CommandHandlerImpl : public CommandHandler
         {
             return CHIP_NO_ERROR;
         }
-        ReturnErrorOnFailure(RollbackResponse());
+        // Here the err from previous step is returned intentionally if RollBackResonse fails due to
+        // running out of packet buffers i.e due to CHIP_ERROR_NO_MEMORY in AllocateBuffer() and
+        // fails to set the mRollBackBackupValid.
```

Comment:
I don't think we should be too specific about the details and do something more along the lines of:
```suggestion
        // `err` from the previous step is returned intentionally if RollBackResponse fails.
        // RollBackResponse only fails if addResponseFunction() was unable to create a checkpoint,
        // which indicates there is nothing to rollback and that trying addResponseFunction will
        // fail for the same reason. The reason why addResponseFunction failed is what is of
        // interest to the caller.
```